pydicom/dataset.py

```diff
@@ -353,11 +369,13 @@ class Dataset(dict):
     """
     indent_chars = "   "
 
-    def __init__(self, *args, **kwargs):
+    def __init__(
+        self, *args: Dict[BaseTag, _DatasetValue], **kwargs: object
```

Comment:
I think it may belong to all args, but we only use the first anyway. This may be better off being an explicit parameter declaration, I'm not really sure why it uses args/kwargs when there's really only two parameters. But changing it would be breaking...

Something like this would be nice
```python
DatasetType = Dict[BaseTag, _DatasetValue]

def __init__(
    self, 
    d: DatasetType, 
    file_meta: Optional[DatasetType] = None, 
    parent: Optional[DatasetType] = None
) -> None
    ...
```

Reply:
[Here's the section in PEP484](https://www.python.org/dev/peps/pep-0484/#arbitrary-argument-lists-and-default-argument-values)

Reply:
Would it be breaking, though? It wouldn't change the usage, if we use explicit parameters - maybe I'm missing something...

Reply:
Hmm, the parameter change from `parent_encoding` to `parent` would be, as would be adding `file_meta` (obviously). But initialising a new `Dataset` or one from a `dict` would remain the same.

Reply:
Right, that is what I meant.

Reply:
Haha, sorry I just realised that's not what you meant.

To make it not breaking we'd have to replicate something like
```python
# These should both work
ds = Dataset()
ds = Dataset({}, 2, 3, ..., {'a': 1, 'b': 2, ...})
```
Which I'm not sure can be done

Reply:
Hm, ok - probably better to leave it for now...